src/acceptance-test/java/de/zalando/aruha/nakadi/webservice/PartitionsControllerAT.java

```diff
@@ -141,11 +142,16 @@ public void whenGetPartitionAndWriteMessageThenOffsetInPartitionIsIncreased() th
 
     private void validateOffsetIncreasedBy(final Map<String, String> partitionInfoBefore,
                                            final Map<String, String> partitionInfoAfter, final long delta) {
-        final long offsetBefore = Long.parseLong(partitionInfoBefore.get("newest_available_offset"));
-        final long offsetAfter = Long.parseLong(partitionInfoAfter.get("newest_available_offset"));
+        final long offsetBefore = getNewestOffsetAsLong(partitionInfoBefore);
+        final long offsetAfter = getNewestOffsetAsLong(partitionInfoAfter);
         assertThat(offsetAfter, is(offsetBefore + delta));
     }
 
+    private Long getNewestOffsetAsLong(final Map<String, String> partitionInfo) {
+        final String offset = partitionInfo.get("newest_available_offset");
+        return Cursor.BEFORE_OLDEST_OFFSET.equals(offset) ? -1 : Long.parseLong(offset);
```

Comment:
Since I had some hard time to understand this, maybe we should just add some comment in the lines of "Handles the case of empty partition, when the current offset would be "BEGIN".